packages/iTwinUI-react/src/core/Table/Table.test.tsx

```diff
@@ -361,62 +361,100 @@ it('should handle checkbox clicks', async () => {
 it('should handle row clicks', async () => {
   const onSelect = jest.fn();
   const onRowClick = jest.fn();
+  const user = userEvent.setup();
+  const data = mockedData(5);
+
   const { container, getByText } = renderComponent({
     isSelectable: true,
     onSelect,
     onRowClick,
+    data,
   });
 
   expect(screen.queryByText('Header name')).toBeFalsy();
   const rows = container.querySelectorAll('.iui-table-body .iui-row');
-  expect(rows.length).toBe(3);
-
-  const user = userEvent.setup();
+  expect(rows.length).toBe(5);
 
-  // Shift click test #1
+  // Shift click special case test #1
   // By default, when no row is selected before shift click, start selecting from first row to clicked row
-  await user.keyboard('[ShiftLeft>]'); // Press Shift (without releasing it)
-  await user.click(getByText(mockedData()[1].name)); // [shiftKey: true]
+  await user.keyboard('[ShiftLeft>]'); // Hold Shift
+  await user.click(getByText(data[2].name));
 
   expect(rows[0].classList).toContain('iui-selected');
   expect(rows[1].classList).toContain('iui-selected');
-  expect(rows[2].classList).not.toContain('iui-selected');
+  expect(rows[2].classList).toContain('iui-selected');
+  expect(rows[3].classList).not.toContain('iui-selected');
   expect(onSelect).toHaveBeenCalledTimes(1);
   expect(onRowClick).toHaveBeenCalledTimes(1);
 
-  await userEvent.click(getByText(mockedData()[1].name)); // Deselect
+  await userEvent.click(getByText(data[1].name)); // Deselect; lastSelectedRowId = undefined -> 1
```

Comment:
Should lastSelectedRowId be 1 if it's deselected? Previously this comment was on "Reselect"

Reply:
Earlier, only selecting updating `lastSelectedRowId`. Now even deselecting does that. So I included this to test this special case.

E.g. After deselecting `1.2.2` and then shift clicking on `1.1`, selection starts from `1.2.2` and not `1.2.4`

[vokoscreenNG-2022-10-21_14-43-58.webm](https://user-images.githubusercontent.com/45748283/197267645-30046366-14d8-451d-863b-c76890a25f33.webm)

Reply:
This feels weird to me. If I deselect something, why would I want to then select it?

Reply:
That's true. I tried to base it off of other examples. When doing the same in Windows Explorer, it does the same.

Example
* Click `packages`
* Ctrl click `CODE OF CONDUCT`
* Ctrl click `packages`
* Shift click `.vscode`

[Video.webm](https://user-images.githubusercontent.com/45748283/198062006-41dba5f7-2f1b-4c24-904c-6016867e8025.webm)

Reply:
Ah, okay